### PR TITLE
add a url_preview_ip_range_whitelist config param

### PR DESCRIPTION
**Pull request by ara4n**
...so we can whitelist the matrix.org IP space

### CODE REVIEW

File: synapse/http/client.py [hunk 1]
```diff
@@ -380,13 +380,15 @@ def post_urlencoded_get_raw(self, url, args={}):
 class SpiderEndpointFactory(object):
     def __init__(self, hs):
         self.blacklist = hs.config.url_preview_ip_range_blacklist
+        if hasattr(hs.config, "url_preview_ip_range_whitelist"):
+            self.whitelist = hs.config.url_preview_ip_range_whitelist
```
Comment (NegativeMjark) [inline synapse/http/client.py]: You will want to set ``self.whitelist`` even if there isn't a ``url_preview_ip_range_whitelist`` in the config.

Perhaps ``self.whitelist = getattr(hs.config, "url_preview_ip_range_whitelist", None)``?

**Comment by ara4n** [comment]
@NegativeMjark thanks - anything else?

File: synapse/config/repository.py [hunk 2]
```diff
@@ -100,6 +100,11 @@ def read_config(self, config):
                     "to work"
                 )
 
+            if "url_preview_ip_range_whitelist" in config:
+                self.url_preview_ip_range_whitelist = IPSet(
+                    config["url_preview_ip_range_whitelist"]
+                )
```
Comment (NegativeMjark) [inline synapse/config/repository.py]: Would it make sense to store the empty IPSet if the config key is missing?

**Comment by NegativeMjark** [comment]
@ara4n I think either one of my suggestions would make the tests pass. I don't think I have any further comments.

**Comment by ara4n** [comment]
lgtm, thanks for fixing it up for me; apologies that it got lost during the main vec delivery.

**Pull request merged**